__mocks__/mock_node.js

```diff
@@ -33,39 +21,48 @@ function mockAll() {
   };
 }
 
+function enhancedModuleResolve(...modules) {
+  // NOTE: __dirname is where I keep some external mocks
+  const resolver = require('enhanced-resolve').create.sync({
+      conditionNames: [process.env.CONFIG_NAME, 'require', 'node', 'default'],
+      extensions: ['.js', '.json', '.node', '.ts', '.tsx'],
+      modules: [__dirname, 'node_modules'],
+    })
+
+  // NOTE: If we redirect all resolutions, breakpoints stop working
+  // Just redirect my own packages (and whichever external API's
+  // I want to have mocked) and everything works correctly
+  const toResolve = ['@thecointech', ...modules]
+  const oldResolve = Module._resolveFilename
+  Module._resolveFilename = (request, parent, isMain) => {
```

Comment:
Some serious hacking of node's internals.  Here we complete the port to conditional exports, by using the enhanved-resolver Node can use conditional exports (like webpack & jest)

Reply:
A bit scary

Reply:
if it works once, it will work forever!!!